src/vm/i386/cgenx86.cpp

```diff
@@ -299,7 +299,12 @@ void TransitionFrame::UpdateRegDisplay(const PREGDISPLAY pRD)
 
     MethodDesc * pFunc = GetFunction();
     _ASSERTE(pFunc != NULL);
+
+#ifdef WIN64EXCEPTIONS
+    UpdateRegDisplayHelper(pRD, 0);
```

Comment:
Hmm, I was looking at it from the amd64 prespective, where the RIP and RSP are extracted in a different way.
The TransitionFrame (resp. the frame types derived from it, since TransitionFrame is an abstract base) is created for transition from managed code. Stack for one of the cases (taken from the PinnedObject test) is below:
```
00007FFFFFFFA000 00007FFFF630460F libcoreclr.so!PreStubWorker + 1935 at /home/janvorli/git/coreclr/src/vm/prestub.cpp:1078
00007FFFFFFFA228                  [PrestubMethodFrame: 00007fffffffa228] System.Reflection.Metadata.dll!System.Reflection.Internal.FileStreamReadLightUp+NativeMethods.ReadFileModern(System.Runtime.InteropServices.SafeHandle, Byte*, Int32, Int32 ByRef, IntPtr)
00007FFFFFFFA270 00007FFFF61EC9BC libcoreclr.so!ThePreStub + 92 at /home/janvorli/git/coreclr/bin/obj/Linux.x64.Debug/src/vm/wks/theprestubamd64.S:2
00007FFFFFFFA360 00007FFF7D7AC7F2 System.Reflection.Metadata.dll!System.Reflection.Internal.FileStreamReadLightUp.TryReadFile(System.IO.Stream, Byte*, Int64, Int32) + 274 [/home/janvorli/git/corefx/src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/FileStreamReadLightUp.cs @ 111]
```
`ThePreStub` asm helper creates a data structure called transition block on the stack. This block contains callee saved and argument registers. The transition block is created right below the return address to the managed caller of the `ThePreStub`. `The PrestubMethodFrame` (derived from the `TransitionFrame`) is created in the `PreStubWorker` and it contains a pointer to the transition block. 
On x64, the RIP is set to the return address to the managed caller and the RSP is set to the end of the transition block plus sizeof(void*) (skipping the return address).
So, these values correspond to the frame of the managed caller of the `ThePreStub`. 
That means that on x86 Linux, the ESP should be the one before the arguments were pushed to be consistent with the cases when we unwind to a managed frame from another managed frame directly. Otherwise it seems that the next unwind would fail if the managed function that called the `ThePreStub` had stack arguments.

Now I also wonder why we extract the pRD->pCurrentContext->Eip and pRD->pCurrentContext->Esp below in a different way than on amd64 (using pRD->PCTAddr instead). 
What I mean, why don't we do it like this:
```
pRD->pCurrentContext->Eip = GetReturnAddress();
pRD->pCurrentContext->Esp = GetSP() + cbStackPop;
```